qa/smoke-test-http/src/test/java/org/elasticsearch/http/snapshots/RestGetSnapshotsIT.java

```diff
@@ -176,7 +182,14 @@ public void testSortAndPaginateWithInProgress() throws Exception {
             inProgressSnapshots.add(AbstractSnapshotIntegTestCase.startFullSnapshot(logger, repoName, snapshotName, false));
         }
         AbstractSnapshotIntegTestCase.awaitNumberOfSnapshotsInProgress(logger, inProgressCount);
-
+        AbstractSnapshotIntegTestCase.awaitClusterState(
```

Comment:
Can you add a comment here on why we need to wait for this instead.

Also, in the PR description, you mention that this makes the sorting stable, can you elaborate on why that is as well?

Reply:
It's copied from https://github.com/elastic/elasticsearch/blob/master/test/framework/src/main/java/org/elasticsearch/snapshots/AbstractSnapshotIntegTestCase.java#L103.

The approach is used in other snapshots tests too, e.g. https://github.com/elastic/elasticsearch/blob/master/server/src/internalClusterTest/java/org/elasticsearch/snapshots/GetSnapshotsIT.java#L42

I can also copy the comment from there
```
Large snapshot pool settings to set up nodes for tests involving multiple repositories that need to have enough
threads so that blocking some threads on one repository doesn't block other repositories from doing work
```

The large pool makes sure that all the shards for `test-index-2` get blocked without blocking `test-index-1`. So we can just wait until `test-index-1` shards have been successfully snapshotted. After that the snapshot in progress reaches a stable state and we can check that the pagination works.

Reply:
I'm not sure this is correct yet. We only wait for `test-index-1` to complete now, but how does that guarantee that `test-index-2` is actually blocked? I suppose in practice the timing works out that way, but it would be nice to have a cleaner and actually deterministic solution here as I can see this still failing albeit at a lower rate than before.

Reply:
@original-brownbear :+1: I've added a check on `test-index-2` to make sure that it's blocked on first shard in the `INIT` state with the rest being `QUEUED`